htdocs/product/class/product.class.php

```diff
@@ -1221,6 +1221,41 @@ public function update($id, $user, $notrigger = false, $action = 'update', $upda
 					}
 				}
 
+				if (!$this->hasbatch() && $this->oldcopy->hasbatch()) {
+					// Selection of all product stock mouvements that contains batchs
+					$sql = 'SELECT * FROM '.MAIN_DB_PREFIX.'stock_mouvement sm';
```

Comment:
Why do you analyze stock movement table ?
If we want to clean the stock for each serial number, we should loop on the table of stock for serial (so llx_product_batch) instead ?  The table stock_movement is not the reflect of the stock and depending on how the stock was initialized may not contains all delta transactions from 0 bu may be only from a certain date. To be sure to have the real stock of each lot to clean, the reliable table is llx_product_batch.